### ISSUE TITLE
[Menu button] Esc key on expanded menu with tooltip closes menu not tooltip

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3049]
## Current behavior

In story [Menu button With Tooltips](https://storybook.sanomalearning.design/?path=/story/actions-menu-menu-button--with-tooltips) cursor users that expand menu and after that make tooltip visible cannot dismiss tooltip with `esc` key. `Esc` key collapses expanded menu, but tooltip stays visible. It disappears only after moving cursor out of menu button. In some cases tooltip can obscure menu items, so `esc` option for dismissing it would be helpful.
This behavior may be problematic for people with limited dexterity, shaking hands and people who use magnifiers/zoom attached to their cursor.

https://github.com/user-attachments/assets/602f7d5f-5ec2-45c4-9860-c1bd6e7248b3

## Suggested solution

It's recommended to add mechanism that checks if tooltip is visible and all the time it's visible it make sure that `esc` dismisses the tooltip. 
So in situation in which tooltip is presented on expanded menu `esc` key should in first step dismiss tooltip, and second use of `esc` should collapse expanded menu.

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS

**Comment by jpzwarte** [comment]
Fixed in #3344
